### PR TITLE
minor adjustments and tests for snapshot creation refactor

### PR DESCRIPTION
**Pull request by tchak**
@igorT requested changes and test

### CODE REVIEW

**Comment by tchak** [comment]
@igorT I rebased and updated after your refactor

File: packages/ember-data/lib/system/store.js [hunk 1]
```diff
@@ -257,11 +257,11 @@ Store = Service.extend({
 
     @method serialize
     @private
-    @param {DS.Model} record the record to serialize
+    @param {InternalModel} record the record to serialize
```
Comment (sly7-7) [inline packages/ember-data/lib/system/store.js]: should we also change the description from record to internalModel ?

Comment (igorT) [inline packages/ember-data/lib/system/store.js]: :+1:

Comment (igorT) [inline packages/ember-data/lib/system/store.js]: actuallly doesn't take an internalModel

File: packages/ember-data/tests/unit/model/merge-test.js [hunk 2]
```diff
@@ -40,6 +40,43 @@ test("When a record is in flight, changes can be made", function() {
...
 });
 
+test("Make sure snapshot is created at save time not at flush time", function() {
+  expect(5);
+
+  var adapter = DS.Adapter.extend({
+    updateRecord: function(store, type, snapshot) {
+      equal(snapshot.attr('name'), 'Thomas Dale');
+
+      return new Ember.RSVP.Promise(function(resolve, reject) {
```
Comment (igorT) [inline packages/ember-data/tests/unit/model/merge-test.js]: you can just return Ember.RSVP.resolve() here

Comment (tchak) [inline packages/ember-data/tests/unit/model/merge-test.js]: updated

File: packages/ember-data/lib/system/store.js (hunk 1 above)
Comment (tchak) [inline packages/ember-data/lib/system/store.js]: fixed

**Pull request merged**